uqcsbot/text.py

```diff
@@ -28,33 +29,33 @@ def __init__(self, bot: commands.Bot):
         self.bot.tree.add_command(self.scare_menu)
 
     @app_commands.command()
-    @app_commands.describe(message="Input string")
-    async def binify(self, interaction: discord.Interaction, message: str):
+    @app_commands.describe(message="Input string", encoding="Character encoding to use, defaults to UTF-8")
+    async def binify(self, interaction: discord.Interaction, message: str, encoding: Optional[str] = "utf-8"):
         """
-        Converts a binary string to an ascii string or vice versa.
+        Converts a binary string to text or vice versa.
         """
         if not message:
             response = "Please include string to convert."
-        elif set(message).issubset(["0", "1", " "]) and len(message) > 2:
-            string = message
-            if len(string) % 8 != 0:
+        elif set(message).issubset(["0", "1"]) and len(message) > 2:
+            if len(message) % 8 != 0:
                 response = "Binary string contains partial byte."
             else:
-                response = ""
-                for i in range(0, len(string), 8):
-                    n = int(string[i:i+8], 2)
-                    if n >= 128:
-                        response = "Character out of ascii range (0-127)"
-                        break
-                    response += chr(n)
+                decoded_message = bytearray()
+                for i in range(0, len(message), 8):
+                    n = int(message[i:i+8], 2)
+                    decoded_message.append(n)
+                try:
+                    response = decoded_message.decode(encoding)
+                except UnicodeDecodeError as e:
+                    response = e.reason
```

Comment:
Maybe add a beginning description to the error like `f"Could not convert to text: {e.reason}"`, but this is just my personal opinion. As long as you're consistent it's alright.

Reply:
I think the no-preamble option is slightly more consistent, considering our manual "partial byte" errors also have no preamble. As an example, trying to decode `10001001` (or any single byte starting with `10`) gives the error message "invalid start byte", which I think is clear enough.